### DISCUSSION TITLE
Chat history error

### DISCUSSION THREAD

**Discussion by Mioooooo** [vanna-ai/vanna#428] [Q&A]
![image](https://github.com/vanna-ai/vanna/assets/37464936/d2fee517-fa77-4fe9-94c3-6f376a027b33)
When I click on chat history, this error page always appears. What should I do?

**Comment by zainhoda** [comment]
We need some more details about your setup, deployment, the sequence of events that lead to this, etc

**Reply by Molrn** [reply]
I have been dealing with that error myself, it comes from the fact that no figure is generated, so no value is set for `fig_json` in the cache. But when the conversation is retrieved from the history, fig_json is required.   
# Reproduction  
```python
from vanna.flask import VannaFlaskApp
from vanna.remote import VannaDefault


class VannaModel(VannaDefault):

    def should_generate_chart(self, df) -> bool:
        return False


vn = VannaModel(model='chinook', api_key="")
vn.connect_to_sqlite('https://vanna.ai/Chinook.sqlite')
VannaFlaskApp(vn).run()
```
1. execute the previous code
2. Open the flask app
3. ask any question
4. reload the page for the question to appear in the history
5. click on the question

result:  
![image](https://github.com/vanna-ai/vanna/assets/76754000/7d8a6b94-ad6f-4579-a645-07bb123cc75d)

**Comment by Mioooooo** [comment]
> We need some more details about your setup, deployment, the sequence of events that lead to this, etc

I don't think I set it up. I just asked a simple question. After asking the question, I clicked on the session page to check, and it prompted the error.

**Reply by Mioooooo** [reply]
![image](https://github.com/vanna-ai/vanna/assets/37464936/e26c2978-a795-4f89-b083-307ff88975f3)
When I refresh the page, click on the history session
![image](https://github.com/vanna-ai/vanna/assets/37464936/6478d2c9-efbe-4372-9b13-6b0d9a984e90)

**Comment by AI-Mart** [comment]
the same error

**Comment by shanjia-sdu** [comment]
我找到问题的原因了，一下是解决方案，希望官方尽快修复。

vanna/flask/__init__.py

generate_plotly_figure 函数里，return之前添加下面一行
cache.set(id=id, field="fig_json", value=fig_json)

generate_followup_questions函数的else分支里return之前，添加下面一行
cache.set(id=id, field="followup_questions", value=["Followup Questions can be enabled if you set allow_llm_to_see_data=True"])


@zainhoda @AI-Mart @Mioooooo @livenson

**Reply by AI-Mart** [reply]
def generate_plotly_figure(user: any, id: str, df, question, sql):
            chart_instructions = flask.request.args.get('chart_instructions')

            if chart_instructions is not None:
                question = f"{question}. When generating the chart, use these special instructions: {chart_instructions}"

            try:
                code = vn.generate_plotly_code(
                    question=question,
                    sql=sql,
                    df_metadata=f"Running df.dtypes gives:\n {df.dtypes}",
                )
                fig = vn.get_plotly_figure(plotly_code=code, df=df, dark_mode=False)
                fig_json = fig.to_json()

                self.cache.set(id=id, field="fig_json", value=fig_json)

                return jsonify(
                    {
                        "type": "plotly_figure",
                        "id": id,
                        "fig": fig_json,
                    }
                )
            except Exception as e:
                # Print the stack trace
                import traceback

                traceback.print_exc()
                self.cache.set(id=id, field="fig_json", value=fig_json)  # jett
                return jsonify({"type": "error", "error": str(e)})

        def generate_followup_questions(user: any, id: str, df, question, sql):
            if self.allow_llm_to_see_data:
                followup_questions = vn.generate_followup_questions(
                    question=question, sql=sql, df=df
                )
                if followup_questions is not None and len(followup_questions) > 5:
                    followup_questions = followup_questions[:5]

                self.cache.set(id=id, field="followup_questions", value=followup_questions)

                return jsonify(
                    {
                        "type": "question_list",
                        "id": id,
                        "questions": followup_questions,
                        "header": "Here are some potential followup questions:",
                    }
                )
            else:
                self.cache.set(id=id, field="followup_questions",
                               value=[
                                   "Followup Questions can be enabled if you set allow_llm_to_see_data=True"])  # jett
                return jsonify(
                    {
                        "type": "question_list",
                        "id": id,
                        "questions": [],
                        "header": "Followup Questions can be enabled if you set allow_llm_to_see_data=True",
                    }
                )
![image](https://github.com/vanna-ai/vanna/assets/63343847/656d5450-65cb-4360-a8d6-d8412558e2fd)

按你的要求改了，还是报错了

**Reply by Mioooooo** [reply]
@shanjia-sdu 我按照你的要求改，还是报错
